src/mlpack/methods/ann/modelParser.cpp

```diff
@@ -0,0 +1,1132 @@
+/**
+ * @file modelParser.cpp
+ * @author Sreenik Seal
+ *
+ * Implementation of a parser to parse json files containing 
+ * user-defined model details to train neural networks
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <iterator>
+#include <queue>
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/init_rules/glorot_init.hpp>
+#include <mlpack/methods/ann/init_rules/const_init.hpp>
+#include <mlpack/methods/ann/init_rules/he_init.hpp>
+#include <mlpack/methods/ann/init_rules/kathirvalavakumar_subavathi_init.hpp>
+#include <mlpack/methods/ann/init_rules/lecun_normal_init.hpp>
+#include <mlpack/methods/ann/init_rules/nguyen_widrow_init.hpp>
+#include <mlpack/methods/ann/init_rules/oivs_init.hpp>
+#include <mlpack/methods/ann/init_rules/orthogonal_init.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+#include <mlpack/methods/ann/loss_functions/cross_entropy_error.hpp>
+//#include <mlpack/methods/ann/loss_functions/earth_mover_distance.hpp>
+#include <mlpack/methods/ann/loss_functions/kl_divergence.hpp>
+#include <mlpack/methods/ann/loss_functions/mean_squared_error.hpp>
+#include <mlpack/methods/ann/loss_functions/sigmoid_cross_entropy_error.hpp>
+//#include <mlpack/methods/ann/loss_functions/reconstruction_loss.hpp>
+#include <mlpack/methods/ann/activation_functions/softsign_function.hpp>
+#include <mlpack/methods/ann/activation_functions/swish_function.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/layer/base_layer.hpp>
+#include <ensmallen.hpp>
+#include <boost/serialization/serialization.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/archive/xml_iarchive.hpp>
+#include <boost/archive/text_iarchive.hpp>
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/tokenizer.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/foreach.hpp>
+#include <boost/algorithm/string.hpp>
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+using namespace arma;
+using namespace std;
+using namespace boost::property_tree;
+
+/**
+ * Implementation of the entire dataset consisting of the training data
+ * and validation data
+ */
+class Dataset
```

Comment:
Would you mind splitting the file into .hpp and .cpp

Reply:
Sure I will do it. Actually the code has some visible scope of improvement that I am aware of (of course, along with things I am not aware of). I will certainly split it before finally requesting a merge. It might have some redundancy in the include statements as well. I'll clean them up after my exams end (or if I can manage time in between).